sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -906,6 +906,23 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
                         .map(x -> getFieldSchema(x.getType(), x.getName(), namespace))
                         .collect(Collectors.toList()));
             break;
+
+          case "NVARCHAR":
+          case "VARCHAR":
+          case "LONGNVARCHAR":
+          case "LONGVARCHAR":
+            baseType = org.apache.avro.Schema.create(Type.STRING);
```

Comment:
These should better be 'proper' Logical types with the associated size. We should probably align the Avro schema representation with the way other systems (Hive / Spark) represent these types into Avro. For ref http://apache-avro.679487.n3.nabble.com/Standardizing-char-and-varchar-logical-types-td4038622.html
or from the source https://github.com/apache/hive/blob/5d268834a5f5278ea76399f8af0d0ab043ae0b45/serde/src/test/resources/avro-struct.avsc#L11

Keeping the full information in the internal representation is ok but we should not be losing the maxLength (size) on the type information.

Apart of this the rest of the PR looks pretty good. Thanks for working on this!

Reply:
Super interesting conversation from 2017!  This could go either way -- the `"char"` and `"varchar"` logical types that Hive adds are not part of the Avro specification and should be ignored by implementations that don't understand them.  A developer could implement their own "custom" logical types that understand them and do any necessary truncation, but AFAIK, nobody does.

I have a small preference for aligning with the de facto behaviour of Spark and Hive, since this information might be useful as data is sent downstream, and it's easy to ignore.  But I'd be OK either way, it could be added in a later PR as a new feature.

Reply:
building the Hive way into the PR.

Reply:
@iemejia Thanks for the Hive reference. Have updated the code, also added a small test in `SchemaUtilTest`.